src/datastore/postgres-store.ts

```diff
@@ -4053,6 +4052,51 @@ export class PgDataStore
     });
   }
 
+  async getUnlockedAddressesAtBlock(burnBlockHeight: number): Promise<StxUnlockEvent[]> {
+    return this.queryTx(async client => {
+      return await this.internalGetUnlockedAccountsAtHeight(client, burnBlockHeight);
+    });
+  }
+
+  async internalGetUnlockedAccountsAtHeight(
+    client: ClientBase,
+    burnBlockHeight: number
+  ): Promise<StxUnlockEvent[]> {
+    const lockQuery = await client.query<{
+      locked_amount: string;
+      unlock_height: string;
+      block_height: string;
+      locked_address: string;
+      tx_id: Buffer;
+    }>(
+      `
+      SELECT locked_amount, unlock_height, block_height, tx_id, locked_address
+      FROM stx_lock_events
+      WHERE canonical = true AND unlock_height = $1
```

Comment:
I'm not sure this query will always work as intended. There can be multiple burn blocks in between Stacks blocks. For example, given `(stacks block 1, burn block 1)` and `(stacks-block 2, burn block 4)`, if unlock height is at burn block 3, then the functions that calls this using a burn block associated with a stacks block will never return the unlock data at burn block 3.

It probably needs changed to find entries where the burn block is less than or equal to `stackBlock.burn_block_height` and greater than `previousStacksBlock.burn_block_height`.

Reply:
Yes, you're right. Fixed it.

Reply:
Looks like a regression after rebase. The above fix was in commit https://github.com/blockstack/stacks-blockchain-api/commit/0ca105278e062f0f7178881e2977006d25b86f3d